instaloader/structures.py

```diff
@@ -394,9 +393,10 @@ def caption_mentions(self) -> List[str]:
             return []
         # This regular expression is modified from jStassen, adjusted to use Python's \w to
         # support Unicode and a word/beginning of string delimiter at the beginning to ensure
-        # that no email addresses join the list of mentions.
+        # that no email addresses join the list of mentions, and adjusted to ensure usernames
+        # only consisting of numbers are excluded.
```

Comment:
`@433` should match `433`, as that seems to be a legal username (that user exists).